google_guest_agent/network/manager/network_manager_linux.go

```diff
@@ -0,0 +1,268 @@
+// managed by the guest agent.
+type nmGuestAgentSection struct {
+	// Managed indicates whether this configuration is created/managed by the agent.
+	Managed bool
+}
+
+// nmConnectionSection is the connection section of NetworkManager's keyfile.
+type nmConnectionSection struct {
+	// InterfaceName is the name of the interface to configure.
+	InterfaceName string `ini:"interface-name"`
+
+	// ID is the unique ID for this connection.
+	ID string `ini:"id"`
+
+	// ConnType is the type of connection (i.e. ethernet).
+	ConnType string `ini:"type"`
+}
+
+// nmIpv4Section is the ipv4 section of NetworkManager's keyfile.
+type nmIpv4Section struct {
+	// Method is the IP configuration method. Supports "auto", "manual", and "link-local".
+	Method string
+}
+
+// nmIpv6Section is the ipv6 section of NetworkManager's keyfile.
+type nmIpv6Section struct {
+	// Method is the IP configuration method. Supports "auto", "manual", and "link-local".
+	Method string
+}
+
+// nmConfig is a wrapper containing all the sections for the NetworkManager keyfile.
+type nmConfig struct {
+	// GuestAgent is the 'guest-agent' section.
+	GuestAgent nmGuestAgentSection `ini:"guest-agent"`
+
+	// Connection is the connection section.
+	Connection nmConnectionSection `ini:"connection"`
+
+	// Ipv4 is the ipv4 section.
+	Ipv4 nmIpv4Section `ini:"ipv4"`
+
+	// Ipv6 is the ipv6 section.
+	Ipv6 nmIpv6Section `ini:"ipv6"`
+}
+
+// networkManager implements the manager.Service interface for NetworkManager.
+type networkManager struct {
+	// configDir is the directory to which to write the configuration files.
+	configDir string
+}
+
+// init registers this network manager service to the list of known network managers.
+func init() {
+	registerManager(&networkManager{
+		configDir: defaultNetworkManagerConfigDir,
+	}, false)
+}
+
+// Name is the name of this network manager service.
+func (n networkManager) Name() string {
+	return "NetworkManager"
+}
+
+// IsManaging checks if NetworkManager is managing the provided interface.
+func (n networkManager) IsManaging(ctx context.Context, iface string) (bool, error) {
+	// Check the status of network.service. This mainly works on SUSE distributions.
+	res := run.WithOutput(ctx, "systemctl", "status", "network.service")
+	if strings.Contains(res.StdOut, "NetworkManager.service") {
+		return true, nil
+	}
+
+	// Check for existence of nmcli. Without nmcli, the agent cannot tell NetworkManager
+	// to reload the configs for its connections.
+	if _, err := execLookPath("nmcli"); err != nil {
+		if errors.Is(err, exec.ErrNotFound) {
+			return false, nil
+		}
+		return false, fmt.Errorf("error checking for nmcli: %v", err)
+	}
+
+	// Check whether NetworkManager.service is active.
+	if err := run.Quiet(ctx, "systemctl", "is-active", "NetworkManager.service"); err != nil {
+		return false, nil
+	}
+
+	// Use nmcli to check status of provided  interface.
+	res = run.WithOutput(ctx, "nmcli", "-f", "DEVICE,STATE", "dev", "status")
```

Comment:
if you use `-t` you'll get a result CSV like output, a more suitable for computers reading format (ps: it can be combined `-f DEVICE,STATE`).

Reply:
Noted. Will change.